### PR TITLE
fix: raise TypeError in V1Alpha1Lease.from_dict for non-dict spec

### LINKED ISSUE #719: V1Alpha1Lease.from_dict raises AttributeError when spec is not a dict
## Description

`V1Alpha1Lease.from_dict` assumes `dict["spec"]` is always a dictionary and calls `.get("selector", {})` on it. If `spec` is present but has a non-dict value (e.g. a list or string), this raises an `AttributeError` instead of a clean validation error.

## Steps to reproduce

```python
from jumpstarter_kubernetes.leases import V1Alpha1Lease

V1Alpha1Lease.from_dict({"spec": []})
# AttributeError: 'list' object has no attribute 'get'
```

## Expected behavior

Should raise `TypeError`, `ValueError`, or `KeyError` (consistent with other malformed-input handling in the method), or defensively handle the case where `spec` is not a dict.

## Actual behavior

```
AttributeError: 'list' object has no attribute 'get'
```

## Location

`python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/leases.py`, `V1Alpha1Lease.from_dict` -- line `selector_data = dict["spec"].get("selector", {})` needs a type check or defensive `.get()` chain.

### PR DESCRIPTION
**Pull request by raballew**
## Summary
- Add type check in `V1Alpha1Lease.from_dict` to raise `TypeError` when `spec` is not a dict, instead of leaking an `AttributeError` from `.get()`
- Extract `spec` into a local variable to avoid repeated `dict["spec"]` lookups
- Add regression tests covering list, string, integer, None, boolean, and empty dict inputs

Fixes https://github.com/jumpstarter-dev/jumpstarter/issues/719

## Test plan
- [x] `test_from_dict_raises_type_error_when_spec_is_a_list`
- [x] `test_from_dict_raises_type_error_when_spec_is_a_string`
- [x] `test_from_dict_raises_type_error_when_spec_is_an_integer`
- [x] `test_from_dict_raises_type_error_when_spec_is_none`
- [x] `test_from_dict_raises_type_error_when_spec_is_a_boolean`
- [x] `test_from_dict_does_not_raise_type_error_when_spec_is_an_empty_dict`
- [x] Existing `test_leases.py` tests still pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by raballew** [comment]
Probably a bit overengineered as k8s should always returns the right type.

**Review by maboras-rh — CHANGES_REQUESTED** [review]

File: python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/leases.py [hunk 1]
```diff
@@ -37,7 +38,10 @@ class V1Alpha1Lease(JsonBaseModel):
 
     @staticmethod
     def from_dict(dict: dict):
```
Comment (maboras-rh) [inline python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/leases.py]: Shadowing python builtin `dict` - the parameter `def from_dict(dict: dict)` shadows the built-in `dict`. This is pre-existing (not introduced by this PR), so not a blocker, but worth noting as a future cleanup opportunity - renaming to `data` or `raw` would be more conventional.

File: python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/leases.py [hunk 2]
```diff
@@ -37,7 +38,10 @@ class V1Alpha1Lease(JsonBaseModel):
...
-        selector_data = dict["spec"].get("selector", {})
+        spec = dict["spec"]
+        if not isinstance(spec, Mapping):
+            raise TypeError(f"spec must be a dict, got {type(spec).__name__}")
```
Comment (maboras-rh) [inline python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/leases.py]: minor point - error message could include the value - `f"spec must be a dict, got {type(spec).__name__}"` tells you the type but not the value. For debugging malformed API responses, including a `repr` of the value could help. 
```suggestion
            raise TypeError(f"spec must be a dict, got {type(spec).__name__}: {repr(spec)}")
```

File: python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/test_leases.py [hunk 3]
```diff
@@ -152,3 +153,33 @@ def test_lease_from_dict_without_match_labels_name_targeted():
...
+    with pytest.raises(TypeError):
+        V1Alpha1Lease.from_dict({"spec": None})
+
+
+def test_from_dict_raises_type_error_when_spec_is_a_boolean():
+    with pytest.raises(TypeError):
+        V1Alpha1Lease.from_dict({"spec": True})
+
+
+def test_from_dict_does_not_raise_type_error_when_spec_is_an_empty_dict():
```
Comment (maboras-rh) [inline python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/test_leases.py]: The empty-dict test name is misleading - `test_from_dict_does_not_raise_type_error_when_spec_is_an_empty_dict` asserts that a KeyError is raised (because apiVersion, kind, metadata, status are all missing).
The name says "does not raise type error" which is technically true, but a reader might expect the call to succeed.
consider renaming it to something like `test_from_dict_with_empty_spec_raises_key_error_not_type_error` to make the intent clearer.

```suggestion
def test_from_dict_with_empty_spec_raises_key_error_not_type_error():
```

File: python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/leases.py (hunk 1 above)
Comment (raballew) [reply python/packages/jumpstarter-kubernetes/jumpstarter_kubernetes/leases.py]: fixed in the last commit

**Review by maboras-rh — APPROVED** [review]
lgtm, clean, well-scoped fix. all three comments have been addressed
approved from my side

**Review by mangelajo — APPROVED** [review]
